### ISSUE TITLE
Text files sometimes get merged, causing intermediate rows with columns names

### ISSUE THREAD

**Issue by cnowzohour** [mendhak/gpslogger#529]
I have GPSLogger set up to synch once per hour with Dropbox and save text files. However, sometimes two text files seem to get merged, causing intermediate header rows and problems when parsing the files. Example:

```
2017-04-12T08:45:33.000Z,40.735390,-73.987455,62.0,47.0,354.6,0.20880614,10,gps,1.6,0.8,1.8,-37.0,,,
2017-04-12T08:45:49.224Z,40.734938,-73.988099,,19.786,,,0,network,,,,,,,
2017-04-12T08:45:59.000Z,40.735218,-73.987466,58.0,28.0,,0.0,11,gps,1.1,0.8,1.4,-37.0,,,
time,lat,lon,elevation,accuracy,bearing,speed,satellites,provider,hdop,vdop,pdop,geoidheight,ageofdgpsdata,dgpsid,activity
2017-04-12T08:46:10.431Z,40.734938,-73.988099,,19.772,,,0,network,,,,,,,
2017-04-12T08:46:21.000Z,40.735209,-73.987479,57.0,15.0,,0.0,10,gps,1.7,0.9,1.9,-37.0,,,
```

**Comment by mendhak** [comment]
What's your logging interval set to?

**Comment by mendhak** [comment]
Are the files merged locally on the phone as well, or are they merged in Dropbox only?

**Comment by cnowzohour** [comment]
Logging interval is set to 10 seconds. Unfortunately I don't have the phones here at the moment, so I can't check the files on the phones. I can try to get hold of them though.

**Comment by mendhak** [comment]
Yes that'd be very useful - it's to determine if this is a local CSV writing problem or if it's a problem with Dropbox uploading and 'somehow' merging files.

**Comment by cnowzohour** [comment]
I've now checked the local files on the phone, and they are the same as on the Dropbox, i.e. it's probably a problem while writing the CSVs. In the corrupted files there was a total of 4 headers including the (correct) one at the top. There is no obvious pattern in the spacing (neither in terms of number of records or time passed between the extra headers.

**Comment by mendhak** [comment]
Can you post the file here OR email the files to me, gpslogger at mendhak.com, I'd like to have a look at the headers anyway.

How often have you seen these merged files, is it the first time or seen it before?
Have there been app updates when this happened, phone restarts?

**Comment by cnowzohour** [comment]
Just sent you an email with the file.

I haven't been using the app for that long (since mid March 2017), and it's been occuring basically since the beginning (although I only realized now). The first day I've seen it was a log from March 22. It's affecting a large proportion of the CSVs (maybe half?). I don't know about connection with app updates / phone restarts, since I'm not carrying the phones myself.

**Comment by mendhak** [comment]
I had a look at the file you sent.  The only time that header line will be written to file is if Android thinkgs [the file doesn't exist](https://github.com/mendhak/gpslogger/blob/master/gpslogger/src/main/java/com/mendhak/gpslogger/loggers/csv/CSVFileLogger.java#L49).  However it's not consistent, it's happening only once in a while rather than consistently or periodically. 

So I wonder a few things.  

You said phones, plural.  Is this happening on multiple phones, are they all the same make and model?  

What logging folder do you use - is it the default, or some other folder, is it on an external SD card?

Are they being mounted/connected to a laptop at any point, I'm wondering if some kind of disk mount operation locks the storage, at which time Android thinks nothing exists, then attempts to write the file due to that.

**Comment by cnowzohour** [comment]
Yes, it seems to be happening on all (six) phones. They are identical models (Motorola Moto G4 Play). The folder is in default location, i.e. on the phone's device storage. As far as I know they are not mounted, although they might be plugged into a USB port for recharging.

**Comment by mendhak** [comment]
Update, over the past few days I've been trying dozens of ways to get CSV files to write in the same merged ways, just not happening.  I'd wonder if it's a single phone problem but then you have 6.  Will continue further testing. 

Also have a look at v87 in the market, did some renaming of CSV files.

**Comment by cnowzohour** [comment]
Yeah unfortunately I don't know how to reproduce it. I have some more phones (different brands) with the app running, will keep checking if I see this behaviour anywhere else.

**Comment by mendhak** [comment]
\---
*Closing as part of issues cleanup, market release, inactivity or inevitability.  Issue can be reopened if needed, or a new issue created.*
